packages/swapper/src/swappers/osmosis/utils/helpers.ts

```diff
@@ -0,0 +1,256 @@
+}
+
+const txStatus = async (txid: string, baseUrl: string): Promise<string> => {
+  try {
+    const txResponse = await axios.get(`${baseUrl}/txs/${txid}`)
+    if (!txResponse?.data?.codespace && !!txResponse?.data?.gas_used) return 'success'
+    if (txResponse?.data?.codespace) return 'failed'
+  } catch (e) {
+    console.error('Failed to get status')
+  }
+  return 'not found'
+}
+
+export const pollForComplete = async (txid: string, baseUrl: string): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    const timeout = 120000 // 2 mins
+    const startTime = Date.now()
+    const interval = 5000 // 5 seconds
+
+    const poll = async function () {
+      const status = await txStatus(txid, baseUrl)
+      if (status === 'success') {
+        resolve(status)
+      } else if (Date.now() - startTime > timeout) {
+        reject(new Error(`Couldnt find tx ${txid}`))
+      } else {
+        setTimeout(poll, interval)
+      }
+    }
+    poll()
+  })
+}
+
+export const getAtomChannelBalance = async (address: string, osmoUrl: string) => {
+  const osmoResponseBalance = await (async () => {
+    try {
+      return axios.get(`${osmoUrl}/bank/balances/${address}`)
+    } catch (e) {
+      throw new Error(`Failed to get osmo balance: ${e}`)
+    }
+  })()
+  let toAtomChannelBalance = 0
+  try {
+    const { amount } = find(
+      osmoResponseBalance.data.result,
+      (b) => b.denom === symbolDenomMapping.ATOM
+    )
+    toAtomChannelBalance = Number(amount)
+  } catch (e) {
+    console.error('no channel balance')
+  }
+  return toAtomChannelBalance
+}
+
+export const pollForAtomChannelBalance = async (
+  address: string,
+  osmoUrl: string
+): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    const timeout = 120000 // 2 mins
+    const startTime = Date.now()
+    const interval = 5000 // 5 seconds
+
+    const poll = async function () {
+      const balance = await getAtomChannelBalance(address, osmoUrl)
+      if (balance > 0) {
+        resolve(balance.toString())
+      } else if (Date.now() - startTime > timeout) {
+        reject(new Error(`Couldnt find channel balance for ${address}`))
+      } else {
+        setTimeout(poll, interval)
+      }
+    }
+    poll()
+  })
+}
+
+const findPool = async (sellAssetSymbol: string, buyAssetSymbol: string, osmoUrl: string) => {
+  const sellAssetDenom = symbolDenomMapping[sellAssetSymbol as keyof SymbolDenomMapping]
+  const buyAssetDenom = symbolDenomMapping[buyAssetSymbol as keyof SymbolDenomMapping]
+
+  const poolsUrl = osmoUrl + '/osmosis/gamm/v1beta1/pools?pagination.limit=1000'
+
+  const poolsResponse = await (async () => {
+    try {
+      return axios.get(poolsUrl)
+    } catch (e) {
+      throw new Error(`Failed to get pool: ${e}`)
+    }
+  })()
+  const foundPool = find(poolsResponse.data.pools, (pool) => {
+    const token0Denom = pool.poolAssets[0].token.denom
```

Comment:
maybe use same buy/sell token verbiage

Reply:
After looking closer, we don't know at this stage which one is buy or sell, they are 0 and 1 index inside the poolAssets data from the osmosis node